src/platform/qt/AudioDevice.cpp

```diff
@@ -63,12 +63,17 @@ qint64 AudioDevice::writeData(const char*, qint64) {
 	return 0;
 }
 
+bool AudioDevice::isSequential() const {
+	return true;
+}
+
 bool AudioDevice::atEnd() const {
-	if (!m_context->core) {
-		return true;
-	}
+	return bytesAvailable() != 0;
+}
+
+qint64 AudioDevice::bytesAvailable() const {
 	mCoreSyncLockAudio(&m_context->impl->sync);
-	bool available = blip_samples_avail(m_context->core->getAudioChannel(m_context->core, 0)) == 0;
+	int n = blip_samples_avail(m_context->core->getAudioChannel(m_context->core, 0));
 	mCoreSyncUnlockAudio(&m_context->impl->sync);
-	return available;
+	return n * sizeof(mStereoSample);
```

Comment:
Nit: leave the variable named `available` even if you change the type--this is the name I use in other places for the same value.